extensions/wikia/EditPageLayout/js/plugins/SyntaxHighlighter.js

```diff
@@ -0,0 +1,563 @@
+			//the styling of the textbox and the background div must be kept very similar
+			var wpTextbox1Style = window.getComputedStyle(wpTextbox1);
+			var scrollTop = wpTextbox1.scrollTop;
+			var focus = (document.activeElement == wpTextbox1);
+
+			wpTextbox0.dir = wpTextbox1.dir;
+			wpTextbox0.lang = wpTextbox1.lang; //lang determines which font "monospace" is
+			wpTextbox0.style.backgroundColor = wpTextbox1Style.backgroundColor;
+			wpTextbox0.style.border = "1px solid transparent";
+			wpTextbox0.style.boxSizing = "border-box";
+			wpTextbox0.style.color = "transparent"; //makes it look just a little bit smoother
+			wpTextbox0.style.fontFamily = wpTextbox1Style.fontFamily;
+			wpTextbox0.style.fontSize = wpTextbox1Style.fontSize;
+			wpTextbox0.style.lineHeight = "normal";
+			wpTextbox0.style.marginBottom = wpTextbox1Style.marginBottom;
+			wpTextbox0.style.marginLeft = "0";
+			wpTextbox0.style.marginRight = "0";
+			wpTextbox0.style.marginTop = wpTextbox1Style.marginTop;
+			wpTextbox0.style.overflowX = "auto";
+			wpTextbox0.style.overflowY = "scroll";
+			//horizontal resize would look horribly choppy, better to make the user resize the browser window instead
+			wpTextbox0.style.resize = (wpTextbox1Style.resize == "vertical" || wpTextbox1Style.resize == "both" ? "vertical" : "none");
+			wpTextbox0.style.tabSize = wpTextbox1Style.tabSize;
+			wpTextbox0.style.whiteSpace = "pre-wrap";
+			wpTextbox0.style.width = "100%";
+			wpTextbox0.style.wordWrap = "normal"; //see below
+
+			wpTextbox1.style.backgroundColor = "transparent";
+			wpTextbox1.style.border = "1px inset gray";
+			wpTextbox1.style.boxSizing = "border-box";
+			wpTextbox1.style.fontSize = wpTextbox1Style.fontSize; //resolves alignment problems on mobile chrome
+			wpTextbox1.style.lineHeight = "normal";
+			wpTextbox1.style.left = "0";
+			wpTextbox1.style.margin = "0";
+			wpTextbox1.style.overflowX = "auto";
+			wpTextbox1.style.overflowY = "scroll";
+			wpTextbox1.style.padding = "0";
+			wpTextbox1.style.position = "absolute";
+			wpTextbox1.style.resize = wpTextbox0.style.resize;
+			wpTextbox1.style.top = "0";
+			wpTextbox1.style.width = "100%";
+			wpTextbox1.style.wordWrap = "normal"; //overall more visually appealing
+
+			//lock both heights to pixel values so that the browser zoom feature works better
+			wpTextbox0.style.height = wpTextbox1.offsetHeight + "px";
+			wpTextbox1.style.height = wpTextbox0.style.height;
+
+			textboxContainer.style.clear = "both";
+			textboxContainer.style.position = "relative";
+
+			wpTextbox1.parentNode.insertBefore(textboxContainer, wpTextbox1);
+			textboxContainer.appendChild(wpTextbox1);
+			textboxContainer.appendChild(wpTextbox0);
+
+			//changing the parent resets scrollTop to 0 and removes focus, so we have to bring that back
+			wpTextbox0.scrollTop = scrollTop;
+			wpTextbox1.scrollTop = scrollTop;
+			if (focus) wpTextbox1.focus();
+
+			//fix drop-downs in editing toolbar
+			$('.tool-select *').css({zIndex: 5});
```

Comment:
Does z-index need to be set on all the descendants or would it be enough on just parent element?